src/test/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumerTest.java

```diff
@@ -25,96 +31,145 @@
         // Given & When
-        OrdersKafkaConsumer consumerUnderTest = new OrdersKafkaConsumer(new SerializerFactory());
-        Message actualMessage = consumerUnderTest.createRetryMessage(ORDER_RECEIVED_URI);
+        OrdersKafkaConsumer consumerUnderTest =
+                new OrdersKafkaConsumer(new SerializerFactory(), new OrdersKafkaProducer(), new KafkaListenerEndpointRegistry());
+        Message actualMessage = consumerUnderTest.createRetryMessage(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC);
         byte[] actualMessageRawValue    = actualMessage.getValue();
         byte[] expectedMessageRawValue  = EXPECTED_MESSAGE_VALUE.getBytes();
         // Then
         Assert.assertThat(actualMessageRawValue, Matchers.is(expectedMessageRawValue));
     }
 
     @Test
-    public void republishMessageToRetryTopicRunsSuccessfully() throws ExecutionException, InterruptedException, SerializationException {
+    public void republishMessageToRetryTopicRunsSuccessfully()
+            throws ExecutionException, InterruptedException, SerializationException {
+        // Given & When
+        when(serializerFactory.getGenericRecordSerializer(OrderReceived.class)).thenReturn(serializer);
+        when(serializer.toBinary(any())).thenReturn(new byte[4]);
+        doCallRealMethod().when(ordersKafkaConsumer).republishMessageToTopic(anyString(), anyString(), anyString());
+        ordersKafkaConsumer.republishMessageToTopic(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY);
+        // Then
+        verify(ordersKafkaProducer, times(1)).sendMessage(any());
+    }
+
+    @Test
+    public void republishMessageToErrorTopicRunsSuccessfully()
+            throws ExecutionException, InterruptedException, SerializationException {
         // Given & When
-        ordersKafkaConsumer.setChKafkaConsumerGroupMain(chKafkaConsumerGroupMain);
         when(serializerFactory.getGenericRecordSerializer(OrderReceived.class)).thenReturn(serializer);
         when(serializer.toBinary(any())).thenReturn(new byte[4]);
-        doCallRealMethod().when(ordersKafkaConsumer).republishMessageToTopic(anyString(), anyString(), anyString(), any());
-        ordersKafkaConsumer.republishMessageToTopic(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC, ORDER_RECEIVED_TOPIC_RETRY, new OrderProcessingException());
+        doCallRealMethod().when(ordersKafkaConsumer).republishMessageToTopic(anyString(), anyString(), anyString());
+        ordersKafkaConsumer.republishMessageToTopic(ORDER_RECEIVED_URI, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR);
         // Then
-        verify(chKafkaConsumerGroupMain, times(3)).retry(anyInt(), any());
+        verify(ordersKafkaProducer, times(1)).sendMessage(any());
     }
 
     @Test
-    public void republishMessageToErrorTopicRunsSuccessfully() throws ExecutionException, InterruptedException, SerializationException {
+    public void republishMessageSuccessfullyCalledOnServiceException()
```

Comment:
* Do these test method names need to change to reflect the new `RetryableErrorException` exception type?

Reply:
Nice catch. Thanks